neo4j/neo4j_test.go

```diff
@@ -0,0 +1,219 @@
+package neo4jstore
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/modules/neo4j"
+)
+
+var testStore *Storage
+
+// TestMain sets up and tears down the test container
+func TestMain(m *testing.M) {
+	ctx := context.Background()
+
+	// Start a Neo4j test container
+	neo4jContainer, err := neo4j.Run(ctx,
```

Comment:
I like the idea of starting with one container per test, so instead of having a TestMain function, having a function that creates the Neo4j store, and calling it por test would be my choice. It will allow parallel execution without the need of taking care of the test data